### DISCUSSION TITLE
User experience for kata test timeout error

### DISCUSSION THREAD

**Discussion by dramforever** [codewars/codewars.com#3021] [Ideas]
Numerous people have been confused by the test result upon a timeout, wondering e.g. why did every test pass but it counts as a failure.

Moreover this current message is a bit too long and hides the most common reason in the middle of the paragraph:

> Why did my code time out?
>
> Our servers are configured to only allow a certain amount of time for your code to execute. In rare cases the server may be taking on too much work and simply wasn't able to run your code efficiently enough. Most of the time though this issue is caused by inefficient algorithms. If you see this error multiple times you should try to optimize your code further.

It's clear that many users unfamiliar with the timeout error find it hard to understand what's going on.

I couldn't find a thread about this so here is one. What are some things we can do about this?

**Comment by hobovsky** [comment]
Related: https://github.com/codewars/runner/issues/69#issuecomment-840926428

**Reply by dramforever** [reply]
no wonder why i didn't find it
